webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/quicksearch/QuickSearchBackendImpl.scala

```diff
@@ -430,14 +431,13 @@ object QSLdapBackend {
     val returnedAttributes =
       (query.attributes.map(_.ldapName).toSeq ++ Seq(A_OC, A_HOSTNAME, A_NAME, A_UUID, A_PARAMETER_NAME, A_IS_SYSTEM)).distinct
 
-    for {
-      connection <- ldap
-      entries    <- connection.search(rudderDit.BASE_DN, Sub, filter, returnedAttributes*)
-    } yield {
-
-      if (ocFilter.isEmpty || attrFilter.isEmpty) { // nothing to search for in that backend
-        Seq()
-      } else {
+    if (ocFilter.isEmpty || attrFilter.isEmpty) { // nothing to search for in that backend
+      Seq.empty[QuickSearchResult].succeed
+    } else {
+      for {
+        connection <- ldap
+        entries    <- connection.search(rudderDit.BASE_DN, Sub, filter, returnedAttributes*)
+      } yield {
```

Comment:
The for-yield is unnecessary when we have nothing to search for...
This change allows for passing everything which is related to LDAP as `null` for unit tests